---
layout: posts
categories: 
  - ecological genetics
title: RNAseq simulation
tags:
  - aphaenogaster
  - simulations
  - transcriptomics
---

## Research

Generate simulated data for RNAseq analysis pipeline. Couldn't get [DWGSIM](http://sourceforge.net/apps/mediawiki/dnaa/index.php?title=
Whole_Genome_Simulation) to install correctly so used wgsim from [SAMTools](http://sourceforge.net/projects/samtools/files/samtools/)

Options allow sequencing error, mutation rate, and variation in read length.

```
Usage:   wgsim [options] <in.ref.fa> <out.read1.fq> <out.read2.fq>
```

- Typical Illumina error rates ~1.3% forward and 1.7% reverse according to
[SEQanswers](http://seqanswers.com/forums/showthread.php?t=6984)


Test by simulating 1,000 2x100PE RNAseq with 500bp size selected fragments, no
mutations. created test input data file by copying first ~6 contigs from
Pogonomyrmex barbatus [predicted
transcriptome](http://www.antgenomes.org/downloads/)

> ./wgsim -d 500 -s 50 -N 1000 -1 100 -2 200 -r 0 -R 0
Pbar_transcriptome_test.fa out.read1.fq out.read2.fq

Worked perfectly!  Generated 978 reads. fastq sequence identified specifies the
contig that each read came from. number of reads per contig ranged from 85 to
241. note that contigs less 650 bp in length disregarded.


## Helms-Cahan Lab group

Jai practice talk for EcoLunch



