---
layout: posts
categories:
  - ecological genetics
  - computing
title: Bowtie2
tags:
  - Bradyrhizobium
  - genome assembly
---

## System admin

Information for:

  - [configuring login sessions](http://mywiki.wooledge.org/DotFiles)
  - [network configuration](https://help.ubuntu.com/12.04/serverguide/network-configuration.html)
  - managing [users](https://help.ubuntu.com/12.04/serverguide/user-management.html)

## Bradyrhizobium

Seems like I'm a bit behind the times with MUMmer. Better options for multiple genome alignment include

  - [Mugsy](http://mugsy.sourceforge.net/)
  - [progressiveMauve](http://gel.ahabs.wisc.edu/mauve/)
  - [SyMAP](http://www.agcol.arizona.edu/software/symap/)

But first, mapping-based assembly to B. japonicum genome using bowtie2.

Fairly easy to understand...got script running quickly!

    #!/bin/bash -l
    #PBS -l nodes=1:ppn=1,mem=24gb,walltime=6:00:00
    #PBS -m abe

    cd /home/tiffinp/stanton1/B-elkanii-genomes/bowtie2-assemblies

    # Set variables
    refgenome=Bjaponicum_USDA110_ref_genome.fasta
    read1="/home/tiffinp/stanton1/B-elkanii-genomes/fastq-quality-trimmed/IC1_R1_seqtk_trimmed_clipped_stillpaired.fastq"
    read2="/home/tiffinp/stanton1/B-elkanii-genomes/fastq-quality-trimmed/IC1_R2_seqtk_trimmed_clipped_stillpaired.fastq"

    # Align contigs from velvet de novo assembly to Bradyrhizobium japonicum reference genome

    # bowtie build
    bowtie2-build $refgenome Bjaponicum

    # bowtie assemble for each sample

    for sample in IC1 ENC4 EWC3
    do
		    bowtie2 -x Bjaponicum -1 $read1 -2 $read2 --phred33 -S BT2-${sample}.sam --un ${sample}-unaligned
    done

Spot-checked SAM output - looks good, but unaligned reads not being saved to separate file. [Recommendation](http://molecularevolution.org/resources/activities/velvet_and_bowtie_activity#exercise4) to remove unmapped reads using:

    awk '$3!="*"' in.sam > in.filt.sam

Searches file for unmapped read with * in 3rd column. Can also do

  awk '$3=="*"' in.sam > in.unaligned.sam

And remove original file.

Picard. Explain SAM [flags](http://picard.sourceforge.net/explain-flags.html)


## Aphaenogaster

Meeting with NG.

  - transcriptome assembly/analysis
  - future experimental plans based on SHC meeting yesterday
    - why do we need to make composite colonies in lab?
      - JSG: control for maternal effects, development environment
    - how many colonies?
    - 

## Computing

  Useful git reminders from software carpentry [git tutorial](https://github.com/redcurry/git_tutorial/blob/master/tutorial.md)

