---
layout: posts
categories: 
  - ApTranscriptome
  - ApPopGen
title: July 14 - July 18 notes
tags: 
  - aster
  - writing
---

## 2014-07-14

Back to work...

* cleaning-up email
* ddRADseq - plans for future work
* ApTranscriptome manuscript organization

### aster

Posted note on estimating selection coefficients the aster way to [google group](https://groups.google.com/forum/#!topic/aster-analysis-user-group/6UlcfGBWM4s)

An issue that has come up in previous analyses is that traits that are significant in the aster models appear not to be significant when performing the OLS regression for selection analysis. Reviewing the tech reports, I realized that this could be due to the fact that I was looking at the standard errors of the coefficients as reported from the OLS regression, but [following Tech Report #675](http://conservancy.umn.edu/bitstream/11299/56394/1/betaTR.pdf) confidence intervals need to be manually calculated. So, need to re-calculate the standard errors of the betas and see if they still overlap 0. 


## 2014-07-15

- ApTranscriptome manuscript
- aster group


The typical output from a study of differential expression is a list of genes that are up or down-regulated in response to some ecological treatment. To aid in biological interpretation, these genes are often clustered by biological function or pathway. However, these results are inconsistent with the goals of evolutionary ecology studies, which are often to understand the evolutionary potential of a species to environmental change. A list of differentially-expressed genes is ...false positives...misses genes with small changes in expression that may have large phenotypic effects...

In genome-wide association studies...largely abandoned by plant and animal breeders in favor of genomic prediction. In this study, we similarly shift our focus from individual genes 


## 2014-07-16

Working on reactionome manuscript.

Procrastination: probability to become PI using [www.pipredictor.com] with my PubMed IDs (24804446, 24443444, 23741505, 23515909, 22919907, 22615745, 22519772, 21935460, 21380848)

~~~
Your probability to become a PI is: 89%

Your score is greater than that of 95% of PIs in our data, while compared to non-PIs, you outscore 95% of them (i.e., false discovery rate of 5%)
~~~

Oddly, using the [Science website algorithm](http://sciencecareers.sciencemag.org/career_magazine/previous_issues/articles/2014_06_02/caredit.a1400136) my prediction is only 53%


## 2014-07-18

Complete draft of Aphaenogaster reactionome manuscript sent to collaborators!


