---
layout: posts
categories: 
  - ApGxL
  - ApAdaPt
title: Thursday April 17 notes
tags: 
  - ddRADseq
  - null hypothesis
  - Bioanalyzer
---

* Lucia's thesis proposal seminar

* Lab meeting - intro to Mike's paper

### ddRADseq

![Bioanalyzer results from double digested samples]({{ site.url }}/assets/files/Cahan-4-17-2014_DNA 1000_DE20901668_2014-04-17_11-00-58.pdf)

With Andrew's help, got the concentration within selected regions (180 - 220 bp, 270 - 330 bp, 360 - 440 bp).

Ln 1, Ap001, is genomic DNA. Note smear above largest marker and no/very little DNA within assay range.

Ln 2 - 6 are double-digested samples. Looks like digest worked well for all samples as concentration in the 180 - 220 bp window ranged from 0.84 - 3.64 ng/ul. 



### ApTranscriptome
 
With help from Feder

1. Install and run [deconseq](http://deconseq.sourceforge.net/)

  - download [databases of contaminants](ftp://edwards.sdsu.edu:7009/deconseq/db/)
  - open `DeconSeqConfig.pm`
  - change database directory to where databases are stored
  - couldn't get script to work if linked from `\usr\local\bin\` so just added deconseq to path in `.bash_rc`
  - run with command

    deconseq.pl -c 50 -i 95 -f Trinity_cap3_uclust.fasta -id Trinity_cap3_uclust -dbs hsref,bast,vir,arch

  where `-c 50` requires 50% coverage and `-i 95` requires 95% similarity to a match in the contaminant database for a read to be considered a contaminant, and `-dbs` specifies the names (listed in DeconSeqConfig.pm file) of the databases to search as contaminants

* 5,675 contaminants
* 99,861 clean

A random 5 contaminants were all BLAST hits to human or bacteria with greater than 95% identity.

  
### CCThermBe
 
Meet with Grace and performed initial analysis of data. Not even no difference, but almost exactly 50:50 foraging among treatments for both experimental colonies! Great demonstration of null result. 

